pelicun/model/loss_model.py

```diff
@@ -1210,72 +1241,101 @@ def _aggregate_sample(self, sample: pd.DataFrame, columns: list) -> pd.DataFrame
     def _apply_replacement_thresholds(
         self,
         sample: pd.DataFrame,
-        replacement_thresholds: dict,
-        replacement_loss_values: dict,
+        replacement_configuration: (
+            tuple[uq.RandomVariableRegistry, dict[str, float]] | None
+        ),
     ) -> tuple[pd.DataFrame, pd.DataFrame]:
 
-        # If there are no `replacement_thresholds`, simply return.
-        if replacement_thresholds is None or not replacement_thresholds:
+        # If there is no `replacement_configuration`, simply return.
+        if replacement_configuration is None:
             # `exceedance_bool_df` is empty in this case.
             exceedance_bool_df = pd.DataFrame(index=sample.index, dtype=bool)
             return sample, exceedance_bool_df
 
-        # otherwise we initialize it with False
+        replacement_consequence_RV_reg, replacement_ratios = (
+            replacement_configuration
+        )
         exceedance_bool_df = pd.DataFrame(
             False,
             index=sample.index,
-            columns=replacement_thresholds.keys(),
+            columns=replacement_consequence_RV_reg.RV.keys(),
             dtype=bool,
         )
 
-        # If there is no `replacement` in the sample, simply return.
-        if 'replacement' not in sample.columns.get_level_values('loss'):
-            return sample, exceedance_bool_df
+        # Sample replacement consequences from the registry
+        replacement_consequence_RV_reg.generate_sample(len(sample), 'MonteCarlo')
+
+        sample_dvs = replacement_consequence_RV_reg.RV.keys()
+        for sample_dv in sample_dvs:
+            sub_sample = sample.loc[:, sample_dv]
+            if 'replacement' in sub_sample.columns.get_level_values('loss'):
+                # If `replacement` already exists as a consequence,
+                # determine the realizations where it is non-zero.
+                no_replacement_mask = (
+                    ~(sub_sample['replacement'] > 0.00).any(axis=1).values
```

Comment:
Why do we need `any` here? Given a specific dv, do we expect more than one `replacement` column?

Reply:
Yes, it's possible to have more than one `replacement` column.
The columns of `sub_sample` have the following levels: [`loss`, `dmg`, `loc`, `dir`, `uid`]. With `sub_sample['replacement']` we are looking for `replacement` in the `loss` level, and there might be multiple matching columns with differing `dmg` entries.

E.g., collapse -> replacement, irreparable -> replacement.

Reply:
Good point, I missed that.